lib/Slack_web_api.js

```diff
@@ -227,10 +227,11 @@ module.exports = function(bot, config) {
 
         request.post(params, function(error, response, body) {
```

Comment:
I refactored this method a bit to handle the case of an `error` and "bad" status codes separately. This rate limiting `if` would have thrown an unhandled exception if `request.post` called the callback with an error (thanks for catching that unit tests!)